### PR TITLE
Rados robust rebalancing

### PR DESCRIPTION
**Pull request by tintumathew10**
Signed-off-by: tintumathew10 <tmathew@redhat.com>

RADOS - robust rebalancing test scenarios.

logs: 
http://magna002.ceph.redhat.com/ceph-qe-logs/tmathew/cephci-run-ZQHJ4B/
http://magna002.ceph.redhat.com/ceph-qe-logs/tmathew/cephci-run-CUYXJ4/

### CODE REVIEW

File: ceph/rados/utils.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    and raise assertion on failure
+    Args:
+        function: name of the function
+        args: arg list
+        kwargs: arg dict
+    """
+    if function(*args, **kwargs):
+        pass
+    else:
+        raise AssertionError(f"Execution failed at function {function}")
```
Comment (psathyan) [inline ceph/rados/utils.py:29]: Can we move this method to `utils/utility.py`? This is a good addition

File: suites/pacific/rados/tier-2_rados_test_osd_rebalance.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      config:
+        verify_cluster_health: true
+        steps:
+          - config:
+              command: bootstrap
+              service: cephadm
+              base_cmd_args:
+                verbose: true
+              args:
+                registry-url: registry.redhat.io
```
Comment (psathyan) [inline suites/pacific/rados/tier-2_rados_test_osd_rebalance.yaml:23]: Could we remove this key/value?

File: suites/pacific/rados/tier-2_rados_test_osd_rebalance.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                - create
+                - cephfs
+          - config:
+              command: apply
+              service: rgw
+              pos_args:
+                - rgw.1
+              args:
+                placement:
+                  label: rgw
```
Comment (psathyan) [inline suites/pacific/rados/tier-2_rados_test_osd_rebalance.yaml:64]: Do you need rgw service to be deployed?

File: ceph/rados/utils.py (hunk 1 above)
Comment (tintumathew10) [reply ceph/rados/utils.py:29]: will do it in next PR.

File: suites/pacific/rados/tier-2_rados_test_osd_rebalance.yaml (hunk 2 above)
Comment (tintumathew10) [reply suites/pacific/rados/tier-2_rados_test_osd_rebalance.yaml]: done

File: suites/pacific/rados/tier-2_rados_test_osd_rebalance.yaml (hunk 3 above)
Comment (tintumathew10) [reply suites/pacific/rados/tier-2_rados_test_osd_rebalance.yaml]: done

**Review by psathyan — APPROVED** [review]
Looks good to me

**Review by pdhiran — APPROVED** [review]
Looks good

**Pull request merged**